project/Teacher.cs

```diff
@@ -69,5 +75,61 @@ public void ViewSchedule()
                 Console.WriteLine($"- {subject.SubjectName} (ID: {subject.SubjectId})");
             }
         }
+
+        private readonly List<Timeslot> _assignedTimeslots = new();
+
+        public List<Timeslot> AssignedTimeslots => new List<Timeslot>(_assignedTimeslots);
```

Comment:
Should be moved above constructor, and the initializations should happen IN the constructor, not in the fields.